File: db/src/main/java/au/org/ala/mobile/ozatlas/db/LocalDateConverter.java
package au.org.ala.mobile.ozatlas.db;

import org.greenrobot.greendao.converter.PropertyConverter;
import org.threeten.bp.LocalDate;
import org.threeten.bp.format.DateTimeFormatter;

public class LocalDateConverter implements PropertyConverter<LocalDate, String> {

    @Override
    public String convertToDatabaseValue(LocalDate localDate) {
        if (localDate == null) return null;
        return localDate.format(DateTimeFormatter.ISO_LOCAL_DATE);
    }

    @Override
    public LocalDate convertToEntityProperty(String date) {
        if (date == null) return null;
        return LocalDate.parse(date);
    }
}

File: ozatlas/src/main/kotlin/au/org/ala/mobile/ozatlas/greendao/LocalDateConverter.kt
package au.org.ala.mobile.ozatlas.greendao

import org.greenrobot.greendao.converter.PropertyConverter
import org.threeten.bp.LocalDate
import org.threeten.bp.format.DateTimeFormatter

class LocalDateConverter : PropertyConverter<LocalDate, String> {

    override fun convertToDatabaseValue(localDate: LocalDate?): String? = localDate?.format(DateTimeFormatter.ISO_LOCAL_DATE)

    override fun convertToEntityProperty(date: String?): LocalDate? = date?.let { LocalDate.parse(date) }
}